lib/stringify.js

```diff
@@ -100,46 +98,42 @@ var stringify = function stringify( // eslint-disable-line func-name-matching
         objKeys = sort ? keys.sort(sort) : keys;
     }
 
+    if (isArray(obj) && generateArrayPrefix === 'comma') {
+        // obj is array like ['a', 'b', 'c']
+        // objKeys is ['0', '1', '2']
+        // we want prefix=a,b,c
+        if (obj.length > 0) {
+            obj['.comma'] = obj.join(',') || null;
```

Comment:
We can't mutate the array someone's passed in; especially not with a property they might themselves be using.